File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/async/AsyncModule.java
package at.tugraz.tc.cyfile.async;


import java.util.concurrent.Executor;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AsyncModule {

    private final Executor executor;

    public AsyncModule(Executor executor) {
        this.executor = executor;
    }

    @Provides
    @Singleton
    public Executor providesExecutor() {
        return executor;
    }
}

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/async/AsyncModule.kt
package at.tugraz.tc.cyfile.async

import dagger.Module
import dagger.Provides
import java.util.concurrent.Executor
import javax.inject.Singleton

@Module
class AsyncModule(val executor: Executor) {


    @Provides
    @Singleton
    fun providesExecutor(): Executor {
        return executor
    }
}